src/realm/column_type_traits.hpp

```diff
@@ -245,27 +253,26 @@ inline bool value_is_null(const util::Optional<T>& val)
 {
     return !val;
 }
-template <>
```

Comment:
It is generally considered better to use function overloads rather than function template explicit specializations. There are fewer footguns in the former. Let me know if there was some reason to prefer specialization here and I can change it back and adjust the ObjectId case to match.